snapcraft/internal/project_loader/_extensions/gnome-3-28.py

```diff
@@ -0,0 +1,111 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Import types and tell flake8 to ignore the "unused" List.
+from typing import Any, Dict, List  # noqa: F401
+
+from snapcraft.internal import errors
+from ._desktop_common import DesktopCommonExtension
+
+
+class GnomeExtensionClassicConfinementError(errors.SnapcraftError):
+    fmt = "The gnome extension doesn't support classic confinement."
+
+
+class Gnome_3_28Extension(DesktopCommonExtension):
+    """The Gnome extension.
+    This extension is to be used by applications that require GTK+.
+    Examples might include productivity applications or utilities.
+    Note that this extension does not support classically-confined snaps at this time.
+    """
+
+    supported_bases = ("core18",)
+
+    def __init__(self, yaml_data: Dict[str, Any]) -> None:
+        """Create a new GnomeExtension.
+        Note that this extension does not support classic snaps.
+        :param dict yaml_data: Loaded snapcraft.yaml data.
+        """
+
+        super().__init__(yaml_data)
+
+        if yaml_data.get("confinement") == "classic":
+            raise GnomeExtensionClassicConfinementError()
+
+        platform_snap = ""
+        base = yaml_data.get("base")
+        after_dependencies = []  # type: List[str]
+        dependency_part = {}  # type: Dict[str, Any]
+        if base is not None:
+            if base == "core18":
+                platform_snap = "gnome-3-28-1804"
+
+        layout = {
+            "layout": {
+                "/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/webkit2gtk-4.0": {
+                    "bind": "$SNAP/gnome-platform/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/webkit2gtk-4.0"
+                },
+                "/usr/share/xml/iso-codes": {
+                    "symlink": "$SNAP/gnome-platform/usr/share/xml/iso-codes"
+                },
+            }
+        }  # type: Dict[str, Any]
+
+        self.root_snippet = {
+            **self.root_snippet,
+            **layout,
+            "plugs": {
+                **self.plugs,
+                platform_snap: {
+                    "interface": "content",
+                    "target": "$SNAP/gnome-platform",
+                    "default-provider": "{snap}:{slot}".format(
+                        snap=platform_snap, slot=platform_snap
+                    ),
+                },
+            },
+            "environment": {
+                **self.environment,
+                "SNAP_DESKTOP_RUNTIME": "$SNAP/gnome-platform",
+            },
+        }
+
+        command_chain = self.app_snippet["command-chain"]
+        exec_command = command_chain.pop()
+
+        command_chain = command_chain + [
+            "snap/command-chain/desktop-gnome-specific",
+            exec_command,
+        ]
+
+        self.app_snippet = {
+            **self.app_snippet,
+            "command-chain": command_chain,
+            "adapter": "full",
```

Comment:
What does that mean nowadays? I thought adapters were gone now that we have the command-chain.

Reply:
Missing a response here.

Reply:
This is from the glib example that I used as inspiration. I do not know the state of adaptors and their existence or not. @sergiusens or @cmatsuoka do you know whether this is still needed or should I drop the `adapter` setting?

Reply:
command-chain arrived a couple of months after we did the 3.0 release, so we sticked to "legacy" as an adapter by default.

We found a couple of scenarios that broke classic confinement (mostly when using shebang/interpreters) so the transition needs a bit more work on our side to complete it.

Reply:
We're talking about a extension being created just now. We shouldn't release this with legacy code that we know is obsolete and needs to be removed as people will start trusting on that. The behavior on any extensions we create need to be supported forever inside the same base.

Reply:
This is why we are forcing the extension to use the new code (it is not the default yet), `adapter: full` makes use of `command-chain` and the vanilla `command` passed on from `snapcraft.yaml` to `snap.yaml`.
Once we make this the default behavior (`adapter: full`) the `adapter` keyword can just go away.

Reply:
I must be missing something. "adapter: full" certainly doesn't sound like "use **no** adapter".

Reply:
Changed to `adapter: none` and added the variable setting to the existing command chain, as discussed with @sergiusens.